esss_fix_format/cli.py

```diff
@@ -30,24 +30,57 @@ def should_format(filename):
     return any(fnmatch(os.path.split(filename)[-1], p) for p in PATTERNS)
 
 
+_created_processes = []
+
+
 @click.command()
 @click.argument('files_or_directories', nargs=-1, type=click.Path(exists=True,
                                                                   dir_okay=True, writable=True))
 @click.option('-k', '--check', default=False, is_flag=True,
-              help='check if files are correctly formatted')
+              help='Check if files are correctly formatted.')
 @click.option('--stdin', default=False, is_flag=True,
-              help='read filenames from stdin (1 per line)')
+              help='Read filenames from stdin (1 per line).')
 @click.option('-c', '--commit', default=False, is_flag=True,
-              help='use modified files from git')
-def main(files_or_directories, check, stdin, commit):
+              help='Use modified files from git.')
+@click.option('--git-hooks', default=False, is_flag=True,
+              help='Add git pre-commit hooks to the repo in the current dir.')
+def main(files_or_directories, check, stdin, commit, git_hooks):
     """Fixes and checks formatting according to ESSS standards."""
 
+    if git_hooks:
+        from esss_fix_format.hook_utils import install_pre_commit_hook
+        install_pre_commit_hook()  # uses the current directory by default.
+        return
+
     formatter = []
 
     def format_code(code_to_format):
         if not formatter:
+            # Note: before actually starting a new process, make sure the previous have been
```

Comment:
Please move this code to a separate function, something like `_wait_current_processes(timeout=0.05)` or so. You can then change the comment to the function's docstring.